lib/compiler.js

```diff
@@ -186,39 +186,82 @@ Compiler.prototype.visitTag = function (tag) {
  * Also, `virtual-dom` handles properties differently.
  */
 
+// attribute -> property transformations
+const ATTRS_TO_PROPS = {
+  'class': 'className',
+  'for': 'htmlFor',
+  'http-equiv': 'httpEquiv',
+}
+const LOWERCASE_ATTRS = new Set([
+  'acceptCharset',
+  'accessKey',
+  'allowFullScreen',
+  'allowTransparency',
+  'cellPadding',
+  'cellSpacing',
+  'colSpan',
+  'contentEditable',
+  'contextMenu',
+  'crossOrigin',
+  'dateTime',
+  'formAction',
+  'formEncType',
+  'formMethod',
+  'formNoValidate',
+  'formTarget',
+  'frameBorder',
+  'marginHeight',
+  'marginWidth',
+  'maxLength',
+  'mediaGroup',
+  'noValidate',
+  'readOnly',
+  'rowSpan',
+  'tabIndex',
+  'useMap',
+])
+for (const a of LOWERCASE_ATTRS) {
+  ATTRS_TO_PROPS[a.toLowerCase()] = a
+}
+
+function dashedToCaps(s) {
+  return s[1].toUpperCase()
```

Comment:
the problem with that is that i wouldn't be able to pass the function directly to `.replace()`, which then defeats the purpose of extracting it in the first place (getting the linter to shut up about creating functions inside of loops). But you're right, it's unreadable garbage and I don't actually care about compile speed micro-optimizations, so I just extracted the whole `dashedToCamel` function in https://github.com/tdumitrescu/virtual-jade/pull/12/commits/0a1e17d4671f953073ee54cbb7f35e5d950c399f